Assets/Scripts/Battlefield/CreatureScripts/AAbilitySO.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace SwordAndBored.Battlefield.CreaturScripts { 
+    public abstract class AAbilitySO : ScriptableObject
+    {
+        public enum ActionTypes { Action, BonusAction, Reaction };
+
+        public string AttackName;
+
+        public ActionTypes TypeOfActionRequired;
+
+
+        public abstract void Initialize(GameObject obj);
+        public abstract void TriggerAbility(GameObject target);
+
+        public int Roll(int dice)
+        {
+            return Random.Range(1, dice);
+        }
+
+        public int Roll()
+        {
+            return Roll(20);
```

Comment:
Are we using the dice mechanism still? i thought we abandoned the DND method with d20's and the like?

Reply:
Ima be honest, I just forgot to remove those methods.  My b